### PR TITLE
filogic: add support for ASUS RT-AX59U

### PR DESCRIPTION
**Pull request by liushiyou006**
Hardware
-------------------
SOC:   MediaTek MT7986
RAM:   512MB DDR4
FLASH: 128MB SPI-NAND (Winbond W25N01GV)
WIFI:  Mediatek MT7986 DBDC 802.11ax 2.4/5 GHz
ETH:   MediaTek MT7531 Switch
UART:  3V3 115200 8N1 (Pinout silkscreened / Do not ocnnect VCC)

Installation
----------------------------------------------------------- 
Install openwrt image:
1. Download the OpenWrt initramfs image. 
  Copy the image to a TFTP server reachable at 192.168.1.70/24. Rename the image to rtax59u.bin.

2. Connect the PC with TFTP server to the RT-AX59U. 
   Set a static ip on the ethernet interface of your PC.
  (ip address: 192.168.1.70, subnet mask:255.255.255.0) 
   Conect to the serial console, interrupt the autoboot process by pressing '4' when prompted.

3. Download & Boot the OpenWrt initramfs image.

   $ setenv ipaddr 192.168.1.1 
   $ setenv serverip 192.168.1.70
   $ tftpboot 0x46000000 rtax59u.bin
   $ bootm 0x46000000

4. Wait for OpenWrt to boot. Transfer the sysupgrade image to the device using scp and install using sysupgrade.

   $ sysupgrade -n <path-to-sysupgrade.bin>
---------------------------------------------------------------------------------------------------- 
Revert to stock firmware:
1: Download the rt-ax59u firmware from ASUS official website. Save the firmware to tftp server directory and rename  to RT-AX59U.trx

2: Connect the PC with TFTP server to the RT-AX59U.
   Set a static ip on the ethernet interface of your PC.
     (ip address: 192.168.1.70, subnet mask:255.255.255.0)

3: Conect to the serial console,  power on again,  interrupt the autoboot process by pressing '4' when prompted.
  $: ubi remove linux
  $: ubi remove jffs2
  $: ubi remove rootfs
  $: ubi remove rootfs_data
  $: ubi create linux 0x45fe000
  $: reset
  then the dut will reboot,interrupt the autoboot process by pressing '2' when prompted.
            2: Load System code then write to Flash via TFTP.
             Warning!! Erase Linux in Flash then burn new one. Are you sure?(Y/N)
             $: enter y
  you will see the follow, hit enter directly:
        Input device IP (192.168.1.1) ==:
        Input server IP (192.168.1.70) ==:
        Input Linux Kernel filename (RT-AX59U.trx) ==:

4: wait for the device run up

### CODE REVIEW

**Comment by raenye** [comment]
Besides the `.dts`, don't you need to add the device to some `.mk` makefile in order to get it built?

**Comment by csharper2005** [comment]
Have you managed to build the images for your device using this commit?

**Comment by liushiyou006** [comment]
> Besides the `.dts`, don't you need to add the device to some `.mk` makefile in order to get it built?
> 
> EDIT: #12933 for instance (also `filogic`) changes multiple files.

Forget push some files, already added.

**Comment by liushiyou006** [comment]
> Have you managed to build the images for your device using this commit?

Forget push some files, already added.

File: target/linux/mediatek/dts/mt7986a-asus-rt-ax59u.dts [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+		compatible = "mediatek,eth-mac";
+		reg = <1>;
+		phy-mode = "2500base-x";
+
+		fixed-link {
+			speed = <2500>;
+			full-duplex;
+			pause;
+		};
+	};
```
Comment (hauke) [inline target/linux/mediatek/dts/mt7986a-asus-rt-ax59u.dts:102]: Why does this device have 2 2.5G links active in the SoC? It looks like the switch is only connected via 1 2.5G link.

File: target/linux/mediatek/dts/mt7986a-asus-rt-ax59u.dts [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_system: system {
+			label = "green:system";
+			gpios = <&pio 11 0>;
```
Comment (xavifr) [inline target/linux/mediatek/dts/mt7986a-asus-rt-ax59u.dts:52]: Green led is active low (0 => GPIO_ACTIVE_LOW).

Red and blue leds are on gpios 12 and 13.

**Comment by hauke** [comment]
#13125 is related and adds support for images using a modified U-Boot

**Comment by tkso1997** [comment]
It seems that MAC addresses are changing after every reboot.

**Comment by remittor** [comment]
@liushiyou006 
> $: ubi create linux 0x45fe000

Why did you decide that partition `linux` is 70 MB in size?
70 MB (linux) + 70 MB (linux2) = 140 MB

**Comment by Wiedzmin89** [comment]
@liushiyou006 @hauke 
I'm currently considering buying a router that supports OpenWrt.
Will support for this router be added to the main branch in the near future?
Information about how to install the OpenWrt version from this branch will be appreciated (seems like I missing `initramfs` image)

File: target/linux/mediatek/filogic/base-files/lib/upgrade/platform.sh [hunk 3]
```diff
@@ -42,7 +42,8 @@ platform_do_upgrade() {
 	local board=$(board_name)
 
 	case "$board" in
-	asus,tuf-ax4200)
+	asus,tuf-ax4200|\
+	asus,rt-ax59u)
```
Comment (raenye) [inline target/linux/mediatek/filogic/base-files/lib/upgrade/platform.sh:46]: Please maintain alphabetic order.

File: target/linux/mediatek/filogic/base-files/lib/preinit/10_fix_eth_mac.sh [hunk 4]
```diff
@@ -2,7 +2,8 @@
 
 preinit_set_mac_address() {
 	case $(board_name) in
-	asus,tuf-ax4200)
+	asus,tuf-ax4200|\
```
Comment (raenye) [inline target/linux/mediatek/filogic/base-files/lib/preinit/10_fix_eth_mac.sh:5]: Please maintain alphabetic order.

File: target/linux/mediatek/filogic/base-files/etc/hotplug.d/ieee80211/11_fix_wifi_mac [hunk 5]
```diff
@@ -10,7 +10,8 @@ PHYNBR=${DEVPATH##*/phy}
 board=$(board_name)
 
 case "$board" in
-	asus,tuf-ax4200)
+	asus,tuf-ax4200|\
```
Comment (raenye) [inline target/linux/mediatek/filogic/base-files/etc/hotplug.d/ieee80211/11_fix_wifi_mac:13]: Please maintain alphabetic order.

File: target/linux/mediatek/filogic/base-files/etc/hotplug.d/firmware/11-mt76-caldata [hunk 6]
```diff
@@ -9,7 +9,8 @@ board=$(board_name)
 case "$FIRMWARE" in
 "mediatek/mt7986_eeprom_mt7976_dbdc.bin")
 	case "$board" in
-	asus,tuf-ax4200)
+	asus,tuf-ax4200|\
```
Comment (raenye) [inline target/linux/mediatek/filogic/base-files/etc/hotplug.d/firmware/11-mt76-caldata:12]: Please maintain alphabetic order.

File: target/linux/mediatek/filogic/base-files/etc/board.d/02_network [hunk 7]
```diff
@@ -57,7 +60,8 @@ mediatek_setup_macs()
 	local label_mac=""
 
 	case $board in
-	asus,tuf-ax4200)
+	asus,tuf-ax4200|\
```
Comment (raenye) [inline target/linux/mediatek/filogic/base-files/etc/board.d/02_network:63]: Please maintain alphabetic order.

File: target/linux/mediatek/filogic/base-files/etc/board.d/02_network [hunk 8]
```diff
@@ -11,6 +11,9 @@ mediatek_setup_interfaces()
 	asus,tuf-ax4200)
 		ucidef_set_interfaces_lan_wan "lan1 lan2 lan3 lan4" eth1
 		;;
+	asus,rt-ax59u)
```
Comment (raenye) [inline target/linux/mediatek/filogic/base-files/etc/board.d/02_network:14]: Please maintain alphabetic order.

File: target/linux/mediatek/dts/mt7986a-asus-rt-ax59u.dts [hunk 9]
```diff
@@ -0,0 +1,276 @@
...
+			partition@0 {
+				label = "Bootloader";
+				reg = <0x0 0x400000>;
+				read-only;
+			};
+
+			partition@400000 {
+				label = "UBI_DEV";
+				reg = <0x400000 0x7c00000>;
+
```
Comment (raenye) [inline target/linux/mediatek/dts/mt7986a-asus-rt-ax59u.dts:250]: This blank line should be removed.

**Comment by hauke** [comment]
Support for ASUS RT-AX59U was merged in #13125.

**Pull request closed without merging**